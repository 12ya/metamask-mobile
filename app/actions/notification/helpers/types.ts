import { Notification } from '../../../util/notifications';
import notificationsAction from './constants';

interface PerformSignInRequest {
  type: typeof notificationsAction.PERFORM_SIGN_IN_REQUEST;
}

interface PerformSignInSuccess {
  type: typeof notificationsAction.PERFORM_SIGN_IN_SUCCESS;
  payload: { accessToken: string };
}

interface PerformSignInFailure {
  type: typeof notificationsAction.PERFORM_SIGN_IN_FAILURE;
  payload: { error: string };
}

interface PerformSignOutRequest {
  type: typeof notificationsAction.PERFORM_SIGN_OUT_REQUEST;
}

interface PerformSignOutSuccess {
  type: typeof notificationsAction.PERFORM_SIGN_OUT_SUCCESS;
}

interface PerformSignOutFailure {
  type: typeof notificationsAction.PERFORM_SIGN_OUT_FAILURE;
  payload: { error: string };
}

interface EnableProfileSyncingRequest {
  type: typeof notificationsAction.ENABLE_PROFILE_SYNCING_REQUEST;
}

interface EnableProfileSyncingSuccess {
  type: typeof notificationsAction.ENABLE_PROFILE_SYNCING_SUCCESS;
}

interface EnableProfileSyncingFailure {
  type: typeof notificationsAction.ENABLE_PROFILE_SYNCING_FAILURE;
  payload: { error: string };
}

interface DisableProfileSyncingRequest {
  type: typeof notificationsAction.DISABLE_PROFILE_SYNCING_REQUEST;
}

interface DisableProfileSyncingSuccess {
  type: typeof notificationsAction.DISABLE_PROFILE_SYNCING_SUCCESS;
}

interface DisableProfileSyncingFailure {
  type: typeof notificationsAction.DISABLE_PROFILE_SYNCING_FAILURE;
  payload: { error: string };
}

interface EnableNotificationsServicesRequest {
  type: typeof notificationsAction.ENABLE_NOTIFICATIONS_SERVICES_REQUEST;
}

interface EnableNotificationsServicesSuccess {
  type: typeof notificationsAction.ENABLE_NOTIFICATIONS_SERVICES_SUCCESS;
}

interface EnableNotificationsServicesFailure {
  type: typeof notificationsAction.ENABLE_NOTIFICATIONS_SERVICES_FAILURE;
  payload: { error: string };
}

interface DisableNotificationsServicesRequest {
  type: typeof notificationsAction.DISABLE_NOTIFICATIONS_SERVICES_REQUEST;
}

interface DisableNotificationsServicesSuccess {
  type: typeof notificationsAction.DISABLE_NOTIFICATIONS_SERVICES_SUCCESS;
}

interface DisableNotificationsServicesFailure {
  type: typeof notificationsAction.DISABLE_NOTIFICATIONS_SERVICES_FAILURE;
  payload: { error: string };
}

interface SetFeatureAnnouncementsEnabledRequest {
  type: typeof notificationsAction.SET_FEATURE_ANNOUNCEMENTS_ENABLED_REQUEST;
}

interface SetFeatureAnnouncementsEnabledSuccess {
  type: typeof notificationsAction.SET_FEATURE_ANNOUNCEMENTS_ENABLED_SUCCESS;
}

interface SetFeatureAnnouncementsEnabledFailure {
  type: typeof notificationsAction.SET_FEATURE_ANNOUNCEMENTS_ENABLED_FAILURE;
  payload: { error: string };
}

interface SetSnapNotificationsEnabledRequest {
  type: typeof notificationsAction.SET_SNAP_NOTIFICATIONS_ENABLED_REQUEST;
}

interface SetSnapNotificationsEnabledSuccess {
  type: typeof notificationsAction.SET_SNAP_NOTIFICATIONS_ENABLED_SUCCESS;
}

interface SetSnapNotificationsEnabledFailure {
  type: typeof notificationsAction.SET_SNAP_NOTIFICATIONS_ENABLED_FAILURE;
  payload: { error: string };
}

interface SetParticipateInMetaMetricsRequest {
  type: typeof notificationsAction.SET_PARTICIPATE_IN_META_METRICS_REQUEST;
}

interface SetParticipateInMetaMetricsSuccess {
  type: typeof notificationsAction.SET_PARTICIPATE_IN_META_METRICS_SUCCESS;
}

interface SetParticipateInMetaMetricsFailure {
  type: typeof notificationsAction.SET_PARTICIPATE_IN_META_METRICS_FAILURE;
  payload: { error: string };
}

interface SetMetamaskNotificationsFeatureSeenRequest {
  type: typeof notificationsAction.SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_REQUEST;
}

interface SetMetamaskNotificationsFeatureSeenSuccess {
  type: typeof notificationsAction.SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_SUCCESS;
}

interface SetMetamaskNotificationsFeatureSeenFailure {
  type: typeof notificationsAction.SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_FAILURE;
  payload: { error: string };
}

interface FetchAndUpdateMetamaskNotificationsRequest {
  type: typeof notificationsAction.FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_REQUEST;
}

interface FetchAndUpdateMetamaskNotificationsSuccess {
  type: typeof notificationsAction.FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_SUCCESS;
  payload: { notifications: Notification[] };
}

interface FetchAndUpdateMetamaskNotificationsFailure {
  type: typeof notificationsAction.FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_FAILURE;
  payload: { error: string };
}

interface MarkMetamaskNotificationsAsReadRequest {
  type: typeof notificationsAction.MARK_METAMASK_NOTIFICATIONS_AS_READ_REQUEST;
  payload: { notifications: Notification[] };
}

interface MarkMetamaskNotificationsAsReadSuccess {
  type: typeof notificationsAction.MARK_METAMASK_NOTIFICATIONS_AS_READ_SUCCESS;
}

interface MarkMetamaskNotificationsAsReadFailure {
  type: typeof notificationsAction.MARK_METAMASK_NOTIFICATIONS_AS_READ_FAILURE;
  payload: { error: string };
}

interface CheckAccountsPresenceRequest {
  type: typeof notificationsAction.CHECK_ACCOUNTS_PRESENCE_REQUEST;
  payload: { accounts: string[] };
}

interface CheckAccountsPresenceSuccess {
  type: typeof notificationsAction.CHECK_ACCOUNTS_PRESENCE_SUCCESS;
  payload: { presence: Record<string, boolean> };
}

interface CheckAccountsPresenceFailure {
  type: typeof notificationsAction.CHECK_ACCOUNTS_PRESENCE_FAILURE;
  payload: { error: string };
}

interface UpdateOnChainTriggersByAccountRequest {
  type: typeof notificationsAction.UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST;
  payload: { accounts: string[] };
}

interface UpdateOnChainTriggersByAccountSuccess {
  type: typeof notificationsAction.UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS;
}

interface UpdateOnChainTriggersByAccountFailure {
  type: typeof notificationsAction.UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE;
  payload: { error: string };
}

interface DeleteOnChainTriggersByAccountRequest {
  type: typeof notificationsAction.DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST;
  payload: { accounts: string[] };
}

interface DeleteOnChainTriggersByAccountSuccess {
  type: typeof notificationsAction.DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS;
}

interface DeleteOnChainTriggersByAccountFailure {
  type: typeof notificationsAction.DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE;
  payload: { error: string };
}

interface EnablePushNotificationsRequest {
  type: typeof notificationsAction.ENABLE_PUSH_NOTIFICATIONS_REQUEST;
  payload: { UUIDs: string[] };
}

interface EnablePushNotificationsSuccess {
  type: typeof notificationsAction.ENABLE_PUSH_NOTIFICATIONS_SUCCESS;
}

interface EnablePushNotificationsFailure {
  type: typeof notificationsAction.ENABLE_PUSH_NOTIFICATIONS_FAILURE;
  payload: { error: string };
}

interface DisablePushNotificationsRequest {
  type: typeof notificationsAction.DISABLE_PUSH_NOTIFICATIONS_REQUEST;
  payload: { UUIDs: string[] };
}

interface DisablePushNotificationsSuccess {
  type: typeof notificationsAction.DISABLE_PUSH_NOTIFICATIONS_SUCCESS;
}

interface DisablePushNotificationsFailure {
  type: typeof notificationsAction.DISABLE_PUSH_NOTIFICATIONS_FAILURE;
  payload: { error: string };
}

interface UpdateTriggerPushNotificationsRequest {
  type: typeof notificationsAction.UPDATE_TRIGGER_PUSH_NOTIFICATIONS_REQUEST;
  payload: { UUIDs: string[] };
}

interface UpdateTriggerPushNotificationsSuccess {
  type: typeof notificationsAction.UPDATE_TRIGGER_PUSH_NOTIFICATIONS_SUCCESS;
  payload: { fcmToken: string };
}

interface UpdateTriggerPushNotificationsFailure {
  type: typeof notificationsAction.UPDATE_TRIGGER_PUSH_NOTIFICATIONS_FAILURE;
  payload: { error: string };
}

export type NotificationsActionTypes =
  | PerformSignInRequest
  | PerformSignInSuccess
  | PerformSignInFailure
  | PerformSignOutRequest
  | PerformSignOutSuccess
  | PerformSignOutFailure
  | EnableProfileSyncingRequest
  | EnableProfileSyncingSuccess
  | EnableProfileSyncingFailure
  | DisableProfileSyncingRequest
  | DisableProfileSyncingSuccess
  | DisableProfileSyncingFailure
  | EnablePushNotificationsRequest
  | EnablePushNotificationsSuccess
  | EnablePushNotificationsFailure
  | DisablePushNotificationsRequest
  | DisablePushNotificationsSuccess
  | DisablePushNotificationsFailure
  | SetFeatureAnnouncementsEnabledRequest
  | SetFeatureAnnouncementsEnabledSuccess
  | SetFeatureAnnouncementsEnabledFailure
  | SetSnapNotificationsEnabledRequest
  | SetSnapNotificationsEnabledSuccess
  | SetSnapNotificationsEnabledFailure
  | SetParticipateInMetaMetricsRequest
  | SetParticipateInMetaMetricsSuccess
  | SetParticipateInMetaMetricsFailure
  | SetMetamaskNotificationsFeatureSeenRequest
  | SetMetamaskNotificationsFeatureSeenSuccess
  | SetMetamaskNotificationsFeatureSeenFailure
  | FetchAndUpdateMetamaskNotificationsRequest
  | FetchAndUpdateMetamaskNotificationsSuccess
  | FetchAndUpdateMetamaskNotificationsFailure
  | MarkMetamaskNotificationsAsReadRequest
  | MarkMetamaskNotificationsAsReadSuccess
  | MarkMetamaskNotificationsAsReadFailure
  | CheckAccountsPresenceRequest
  | CheckAccountsPresenceSuccess
  | CheckAccountsPresenceFailure
  | UpdateOnChainTriggersByAccountRequest
  | UpdateOnChainTriggersByAccountSuccess
  | UpdateOnChainTriggersByAccountFailure
  | DeleteOnChainTriggersByAccountRequest
  | DeleteOnChainTriggersByAccountSuccess
  | DeleteOnChainTriggersByAccountFailure
  | UpdateTriggerPushNotificationsRequest
  | UpdateTriggerPushNotificationsSuccess
  | UpdateTriggerPushNotificationsFailure
  | EnableNotificationsServicesRequest
  | EnableNotificationsServicesSuccess
  | EnableNotificationsServicesFailure
  | DisableNotificationsServicesRequest
  | DisableNotificationsServicesSuccess
  | DisableNotificationsServicesFailure;
