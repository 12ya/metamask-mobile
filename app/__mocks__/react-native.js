import { Linking, DeviceEventEmitter } from 'react-native';

Linking.addEventListener = jest.fn();
Linking.removeEventListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeEventListener
  if (typeof callback === 'function') {
    callback();
  }
});
Linking.openURL = jest.fn();
Linking.canOpenURL = jest.fn();
Linking.getInitialURL = jest.fn();

DeviceEventEmitter.addListener = jest.fn();
DeviceEventEmitter.removeListener = jest.fn((event, callback) => {
  // Simulate the behavior of removeListener
  if (typeof callback === 'function') {
    callback();
  }
});

export default { Linking, DeviceEventEmitter };
