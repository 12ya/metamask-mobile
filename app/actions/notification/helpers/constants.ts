enum notificationsAction {
  PERFORM_SIGN_IN_REQUEST = 'notifications/perform_sign_in_request',
  PERFORM_SIGN_IN_SUCCESS = 'notifications/perform_sign_in_success',
  PERFORM_SIGN_IN_FAILURE = 'notifications/perform_sign_in_failure',
  PERFORM_SIGN_OUT_REQUEST = 'notifications/perform_sign_out_request',
  PERFORM_SIGN_OUT_SUCCESS = 'notifications/perform_sign_out_success',
  PERFORM_SIGN_OUT_FAILURE = 'notifications/perform_sign_out_failure',
  ENABLE_PROFILE_SYNCING_REQUEST = 'notifications/enable_profile_syncing_request',
  ENABLE_PROFILE_SYNCING_SUCCESS = 'notifications/enable_profile_syncing_success',
  ENABLE_PROFILE_SYNCING_FAILURE = 'notifications/enable_profile_syncing_failure',
  DISABLE_PROFILE_SYNCING_REQUEST = 'notifications/disable_profile_syncing_request',
  DISABLE_PROFILE_SYNCING_SUCCESS = 'notifications/disable_profile_syncing_success',
  DISABLE_PROFILE_SYNCING_FAILURE = 'notifications/disable_profile_syncing_failure',
  ENABLE_PUSH_NOTIFICATIONS_REQUEST = 'notifications/enable_push_notifications_request',
  ENABLE_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/enable_push_notifications_success',
  ENABLE_PUSH_NOTIFICATIONS_FAILURE = 'notifications/enable_push_notifications_failure',
  DISABLE_PUSH_NOTIFICATIONS_REQUEST = 'notifications/disable_push_notifications_request',
  DISABLE_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/disable_push_notifications_success',
  DISABLE_PUSH_NOTIFICATIONS_FAILURE = 'notifications/disable_push_notifications_failure',
  CHECK_ACCOUNTS_PRESENCE_REQUEST = 'notifications/check_accounts_presence_request',
  CHECK_ACCOUNTS_PRESENCE_SUCCESS = 'notifications/check_accounts_presence_success',
  CHECK_ACCOUNTS_PRESENCE_FAILURE = 'notifications/check_accounts_presence_failure',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST = 'notifications/delete_on_chain_triggers_by_account_request',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS = 'notifications/delete_on_chain_triggers_by_account_success',
  DELETE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE = 'notifications/delete_on_chain_triggers_by_account_failure',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_REQUEST = 'notifications/update_on_chain_triggers_by_account_request',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_SUCCESS = 'notifications/update_on_chain_triggers_by_account_success',
  UPDATE_ON_CHAIN_TRIGGERS_BY_ACCOUNT_FAILURE = 'notifications/update_on_chain_triggers_by_account_failure',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_REQUEST = 'notifications/set_feature_announcements_enabled_request',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_SUCCESS = 'notifications/set_feature_announcements_enabled_success',
  SET_FEATURE_ANNOUNCEMENTS_ENABLED_FAILURE = 'notifications/set_feature_announcements_enabled_failure',
  SET_SNAP_NOTIFICATIONS_ENABLED_REQUEST = 'notifications/set_snap_notifications_enabled_request',
  SET_SNAP_NOTIFICATIONS_ENABLED_SUCCESS = 'notifications/set_snap_notifications_enabled_success',
  SET_SNAP_NOTIFICATIONS_ENABLED_FAILURE = 'notifications/set_snap_notifications_enabled_failure',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_REQUEST = 'notifications/set_metamask_notifications_feature_seen_request',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_SUCCESS = 'notifications/set_metamask_notifications_feature_seen_success',
  SET_METAMASK_NOTIFICATIONS_FEATURE_SEEN_FAILURE = 'notifications/set_metamask_notifications_feature_seen_failure',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_REQUEST = 'notifications/fetch_and_update_metamask_notifications_request',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_SUCCESS = 'notifications/fetch_and_update_metamask_notifications_success',
  FETCH_AND_UPDATE_METAMASK_NOTIFICATIONS_FAILURE = 'notifications/fetch_and_update_metamask_notifications_failure',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_REQUEST = 'notifications/mark_metamask_notifications_as_read_request',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_SUCCESS = 'notifications/mark_metamask_notifications_as_read_success',
  MARK_METAMASK_NOTIFICATIONS_AS_READ_FAILURE = 'notifications/mark_metamask_notifications_as_read_failure',
  DELETE_NOTIFICATION_STATUS_REQUEST = 'notifications/delete_notification_status_request',
  DELETE_NOTIFICATION_STATUS_SUCCESS = 'notifications/delete_notification_status_success',
  DELETE_NOTIFICATION_STATUS_FAILURE = 'notifications/delete_notification_status_failure',
  SET_PARTICIPATE_IN_META_METRICS_REQUEST = 'notifications/set_participate_in_meta_metrics_request',
  SET_PARTICIPATE_IN_META_METRICS_SUCCESS = 'notifications/set_participate_in_meta_metrics_success',
  SET_PARTICIPATE_IN_META_METRICS_FAILURE = 'notifications/set_participate_in_meta_metrics_failure',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_REQUEST = 'notifications/update_trigger_push_notifications_request',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_SUCCESS = 'notifications/update_trigger_push_notifications_success',
  UPDATE_TRIGGER_PUSH_NOTIFICATIONS_FAILURE = 'notifications/update_trigger_push_notifications_failure',
  ENABLE_NOTIFICATIONS_SERVICES_REQUEST = 'notifications/enable_notifications_services_request',
  ENABLE_NOTIFICATIONS_SERVICES_SUCCESS = 'notifications/enable_notifications_services_success',
  ENABLE_NOTIFICATIONS_SERVICES_FAILURE = 'notifications/enable_notifications_services_failure',
  DISABLE_NOTIFICATIONS_SERVICES_REQUEST = 'notifications/disable_notifications_services_request',
  DISABLE_NOTIFICATIONS_SERVICES_SUCCESS = 'notifications/disable_notifications_services_success',
  DISABLE_NOTIFICATIONS_SERVICES_FAILURE = 'notifications/disable_notifications_services_failure',
}

export default notificationsAction;
