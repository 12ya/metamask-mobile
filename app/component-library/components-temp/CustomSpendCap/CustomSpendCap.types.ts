export interface CustomSpendCapProps {
  ticker: string;
  dappProposedValue: string;
  accountBalance: string;
  domain: string;
}
