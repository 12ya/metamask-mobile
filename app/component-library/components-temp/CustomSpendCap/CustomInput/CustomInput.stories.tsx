import React from 'react';
import { View } from 'react-native';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import CustomInput from './CustomInput';
import type { Meta, StoryObj } from '@storybook/react-native';
import {
  CUSTOM_SPEND_CAP_INPUT_TEST_ID,
  CUSTOM_SPEND_CAP_MAX_TEST_ID,
  CUSTOM_SPEND_CAP_INPUT_INPUT_ID,
} from './CustomInput.constants';

// Mock Redux store
const store = createStore(() => ({}));

const withProvider = (Story: React.ComponentType) => (
  <Provider store={store}>
    <View>
      <Story />
    </View>
  </Provider>
);

const meta: Meta<typeof CustomInput> = {
  title: 'Components Temp / CustomSpendCap / CustomInput',
  component: CustomInput,
  decorators: [withProvider],
  argTypes: {
    ticker: { control: 'text' },
    value: { control: 'text' },
    setValue: { action: 'setValue' },
    isInputGreaterThanBalance: { control: 'boolean' },
    setMaxSelected: { action: 'setMaxSelected' },
    isEditDisabled: { control: 'boolean' },
    tokenDecimal: { control: 'number' },
  },
};

export default meta;

type Story = StoryObj<typeof CustomInput>;

export const Default: Story = {
  args: {
    ticker: 'ETH',
    value: '',
    isInputGreaterThanBalance: false,
    isEditDisabled: false,
    tokenDecimal: 18,
  },
};

export const WithValue: Story = {
  args: {
    ...Default.args,
    value: '1.5',
  },
};

export const InputGreaterThanBalance: Story = {
  args: {
    ...Default.args,
    value: '1000',
    isInputGreaterThanBalance: true,
  },
};

export const EditDisabled: Story = {
  args: {
    ...Default.args,
    value: '2.5',
    isEditDisabled: true,
  },
};
