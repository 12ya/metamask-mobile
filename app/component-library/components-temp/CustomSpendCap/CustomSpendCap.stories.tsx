// Third party dependencies.
import React from 'react';
import { storiesOf } from '@storybook/react-native';

// Internal dependencies.
import {
  TICKER,
  ACCOUNT_BALANCE,
  DAPP_PROPOSED_VALUE,
  DAPP_DOMAIN,
} from './CustomSpendCap.constants';
import CustomSpendCap from './CustomSpendCap';

storiesOf('Component Library / CustomSpendCap', module).add('Default', () => (
  <CustomSpendCap
    ticker={TICKER}
    accountBalance={ACCOUNT_BALANCE}
    dappProposedValue={DAPP_PROPOSED_VALUE}
    domain={DAPP_DOMAIN}
  />
));
