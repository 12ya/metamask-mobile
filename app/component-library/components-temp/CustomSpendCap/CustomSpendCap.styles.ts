// Third party dependencies.
import { StyleSheet } from 'react-native';

import { Theme } from '../../../util/theme/models';
/**
 * Style sheet for CustomSpendCap component.
 *
 * @returns StyleSheet object.
 */

const styleSheet = (params: { theme: Theme }) => {
  const { theme } = params;
  return StyleSheet.create({
    container: {
      backgroundColor: theme.colors.background.alternative,
      borderRadius: 5,
      marginHorizontal: 8,
      padding: 12,
    },
    header: {
      flexDirection: 'row',
      justifyContent: 'space-between',
    },
    titleContainer: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    description: {
      color: theme.colors.text.alternative,
      marginHorizontal: 8,
    },
    errorValue: {
      color: theme.colors.error.default,
      marginHorizontal: 8,
      marginBottom: 8,
    },
  });
};

export default styleSheet;
