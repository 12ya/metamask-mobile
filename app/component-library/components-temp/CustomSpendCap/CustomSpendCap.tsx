// Third party dependencies.
import React, { useState, useEffect } from 'react';
import { View } from 'react-native';

// External dependencies.
import { useStyles } from '../../hooks';
import { strings } from '../../../../locales/i18n';
import ButtonLink from '../../components/Buttons/Button/variants/ButtonLink';
import Text, { TextVariants } from '../../components/Texts/Text';
import formatNumber from '../../../util/formatNumber';
import CustomInput from './CustomInput';

// Internal dependencies.
import { CUSTOM_SPEND_CAP_TEST_ID } from './CustomSpendCap.constants';
import { CustomSpendCapProps } from './CustomSpendCap.types';
import customSpendCapStyles from './CustomSpendCap.styles';

const CustomSpendCap = ({
  ticker,
  dappProposedValue,
  accountBalance,
  domain,
}: CustomSpendCapProps) => {
  const { styles } = useStyles(customSpendCapStyles, {});

  const [value, setValue] = useState('');
  const [inputDisabled, setInputDisabled] = useState(true);
  const [maxSelected, setMaxSelected] = useState(false);
  const [defaultValueSelected, setDefaultValueSelected] = useState(false);
  const [
    inputValueHigherThanAccountBalance,
    setInputValueHigherThanAccountBalance,
  ] = useState(false);

  const handlePress = () => {
    setMaxSelected(false);
    setValue(dappProposedValue);
    setDefaultValueSelected(!defaultValueSelected);
    setInputDisabled(!inputDisabled);
  };

  useEffect(() => {
    if (maxSelected) setValue(accountBalance);
  }, [maxSelected, accountBalance]);

  const dappValue = Number(dappProposedValue) - Number(accountBalance);
  const difference = Number(value) - Number(accountBalance);

  useEffect(() => {
    if (Number(value) > Number(accountBalance)) {
      setInputValueHigherThanAccountBalance(true);
    } else {
      setInputValueHigherThanAccountBalance(false);
    }
  }, [value, accountBalance]);

  const MAX_VALUE_SELECTED = strings(
    'contract_allowance.custom_spend_cap.max_value_selected',
    { accountBalance: formatNumber(accountBalance), ticker },
  );
  const NO_SELECTED = strings(
    'contract_allowance.custom_spend_cap.no_value_selected',
    { domain },
  );
  const DAPP_PROPOSED_VALUE_GREATER_THAN_ACCOUNT_BALANCE = strings(
    'contract_allowance.custom_spend_cap.dapp_proposed_value_greater_than_account_balance',
    {
      accountBalance: formatNumber(accountBalance),
      dappValue: formatNumber(dappValue),
      ticker,
    },
  );
  const INPUT_VALUE_GREATER_THAN_ACCOUNT_BALANCE = strings(
    'contract_allowance.custom_spend_cap.input_value_greater_than_account_balance',
    {
      accountBalance: formatNumber(accountBalance),
      difference: formatNumber(difference),
      ticker,
    },
  );

  return (
    <View style={styles.container} testID={CUSTOM_SPEND_CAP_TEST_ID}>
      <View style={styles.header}>
        <Text variant={TextVariants.sBodyMDBold}>Custom spending cap</Text>
        {defaultValueSelected ? (
          <ButtonLink onPress={handlePress} textVariants={TextVariants.sBodyMD}>
            {strings('contract_allowance.custom_spend_cap.edit')}
          </ButtonLink>
        ) : (
          <ButtonLink onPress={handlePress} textVariants={TextVariants.sBodyMD}>
            {strings('contract_allowance.custom_spend_cap.use_default')}
          </ButtonLink>
        )}
      </View>
      <CustomInput
        ticker={ticker}
        setValue={setValue}
        defaultValueSelected={defaultValueSelected}
        setMaxSelected={setMaxSelected}
        inputDisabled={inputDisabled}
        value={value}
      />
      <Text variant={TextVariants.sBodyMD} style={styles.description}>
        {defaultValueSelected
          ? DAPP_PROPOSED_VALUE_GREATER_THAN_ACCOUNT_BALANCE
          : maxSelected
          ? MAX_VALUE_SELECTED
          : inputValueHigherThanAccountBalance
          ? INPUT_VALUE_GREATER_THAN_ACCOUNT_BALANCE
          : NO_SELECTED}
      </Text>
    </View>
  );
};

export default CustomSpendCap;
