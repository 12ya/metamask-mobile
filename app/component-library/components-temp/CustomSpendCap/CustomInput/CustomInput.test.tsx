import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import CustomInput from './CustomInput';
import {
  CUSTOM_SPEND_CAP_INPUT_INPUT_ID,
  CUSTOM_SPEND_CAP_MAX_TEST_ID,
} from './CustomInput.constants';
import { CustomInputProps } from './CustomInput.types';
import { TICKER } from '../CustomSpendCap.constants';

describe('CustomInput', () => {
  let props: CustomInputProps;

  beforeEach(() => {
    props = {
      ticker: TICKER,
      value: '123',
      isInputGreaterThanBalance: false,
      isEditDisabled: false,
      setMaxSelected: jest.fn(),
      setValue: jest.fn(),
      tokenDecimal: 4,
    };
  });

  const renderComponent = () => render(<CustomInput {...props} />);

  it('should render correctly', () => {
    const { toJSON } = renderComponent();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call setMaxSelected when max button is pressed', () => {
    const { getByTestId } = renderComponent();
    fireEvent.press(getByTestId(CUSTOM_SPEND_CAP_MAX_TEST_ID));
    expect(props.setMaxSelected).toHaveBeenCalled();
  });

  it('should update value if input is integer', () => {
    const { getByTestId } = renderComponent();
    fireEvent.changeText(getByTestId(CUSTOM_SPEND_CAP_INPUT_INPUT_ID), '123');
    expect(props.setValue).toHaveBeenCalledWith('123');
  });

  it('should update value if input is decimal and decimal points are less than or equal to tokenDecimal', () => {
    const { getByTestId } = renderComponent();
    fireEvent.changeText(
      getByTestId(CUSTOM_SPEND_CAP_INPUT_INPUT_ID),
      '123.1234',
    );
    expect(props.setValue).toHaveBeenCalledWith('123.1234');
  });

  it('should not update value if input is decimal and decimal points are greater than tokenDecimal', () => {
    const { getByTestId } = renderComponent();
    fireEvent.changeText(
      getByTestId(CUSTOM_SPEND_CAP_INPUT_INPUT_ID),
      '123.1234567',
    );
    expect(props.setValue).not.toHaveBeenCalled();
  });
});
