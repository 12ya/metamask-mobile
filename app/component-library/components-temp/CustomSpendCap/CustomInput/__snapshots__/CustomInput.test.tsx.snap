// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "margin": 10,
      "padding": 10,
    }
  }
  testID="custom-input-test-id"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginHorizontal": 5,
        "width": "90%",
      }
    }
  >
    <TextInput
      keyboardType="numeric"
      multiline={true}
      onChangeText={[Function]}
      placeholder="Enter a number here (DAI)"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
        ]
      }
      value="123"
    />
  </View>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#535A61",
      }
    }
    variant="sBodySM"
  >
    Max
  </Text>
</View>
`;
